haystack/components/embedders/sentence_transformers_document_embedder.py

```diff
@@ -128,7 +140,12 @@ def to_dict(self) -> Dict[str, Any]:
             embedding_separator=self.embedding_separator,
             trust_remote_code=self.trust_remote_code,
             truncate_dim=self.truncate_dim,
+            model_kwargs=self.model_kwargs,
+            tokenizer_kwargs=self.tokenizer_kwargs,
         )
+        if serialization_dict["init_parameters"].get("model_kwargs") is not None:
```

Comment:
Just wondering: might we need something similar for the tokenizer kwargs too?

Reply:
I don't think so, the model_kwargs has to be treated specially mostly due to torch.float types not being serializable. I'm not sure if tokenizer kwargs contains any type of non-serializable values. Taking a quick look [here](https://huggingface.co/docs/transformers/en/main_classes/tokenizer#transformers.PreTrainedTokenizer) I'd say all values could be specified using python default types (e.g. str, dict, list, tuple) so I think it's okay.